chainstate/src/interface/chainstate_interface.rs

```diff
@@ -166,4 +166,31 @@ pub trait ChainstateInterface: Send {
 
     /// Returns true if the initial block download isn't finished yet.
     fn is_initial_block_download(&self) -> Result<bool, ChainstateError>;
+
+    fn pool_exists(&self, pool_id: PoolId) -> Result<bool, ChainstateError>;
```

Comment:
staking_pool_exists()

Pool alone is confusing without context. Same for all other instances here.

Reply:
Good point, will update all staking-related methods exposed in chainstate.